lib/ansible/plugins/inventory/foreman.py

```diff
@@ -166,8 +166,8 @@ def _get_all_params_by_id(self, hid):
         url = "%s/api/v2/hosts/%s" % (self.foreman_url, hid)
         ret = self._get_json(url, [404])
         if not ret or not isinstance(ret, MutableMapping) or not ret.get('all_parameters', False):
-            ret = {'all_parameters': [{}]}
-        return ret.get('all_parameters')[0]
+            ret = []
```

Comment:
When this path is triggered, it will raise an Exception because lists do not have a get() method.  So ```return ret.get('all_parameters')``` will fail.

Reply:
I have cherry-picked the commits from #54333